### PR TITLE
rlp: v3 integration for monorepo packages

### PR DESCRIPTION
**Pull request by ryanio**
This PR integrates rlp v3 into the monorepo, targeted to merge into the `develop` branch.

### CODE REVIEW

**Comment by holgerd77** [comment]
Maybe let's keep this open for at least a few more days respectively 1-2 weeks, have added a "blocked" label for now.

Would be nice if we heard a couple of more voices on the BigInt integration + also see the results from #1652, since this would "finalize" on the BigInt decision. And also no imminent need to rush this in (might then need another 1-2 rounds of rebases though, sorry for that 😬).

**Comment by ryanio** [comment]
no problem, I can keep occasionally keep this up-to-date. it was just good/reassuring to see everything passing :)

**Comment by holgerd77** [comment]
Note: I am just in the process of recreating the `develop` branch with a clean commit history now always using rebases to get on top of the `master` branch, see post in the internal "#BigInt Conversion" channel.

I will close this PR here, have done a branch copy locally as a backup just in case.

I will then delete and repush the `develop` branch, then under a fitting backup name, and repush a "new" `develop` branch for continued development with the existing squashed merge PR commits cherry-picked.

Please reopen this PR towards this branch.

Thanks! 🙂

**Comment by holgerd77** [comment]
Ok, have recreated the `develop` branch, feel free to reopen now.

**Comment by holgerd77** [comment]
Ah, this direct re-opening seems to work (though more of "re-open as a new PR"). That's interesting. 🙂

**Comment by ryanio** [comment]
yeah I didn't want a third PR with this name (haha) so I reopened, rebased on the current develop (no conflicts, that was nice!), then force pushed, seems to have gone well

**Comment by holgerd77** [comment]
Just force-pushed a rebased `develop`.

Sorry, I wonder how much hazzle this will cause, might be best on this round to create a new branch from the new `develop` head and then cherry-pick commits on top one-by-one. We might want to switch to only do these base branch rebases on times where all `develop` work has been merged, this should make things easier.

I guess generally we can merge this branch now? Would also be good if we can do relatively soon, guess things won't get better over time.

**Comment by ryanio** [comment]
yeah, I can fix conflicts if we want to merge this in, will try to get to it soon

**Comment by ryanio** [comment]
I will this pick this up next week and rebase to get it ready for merging (after bigint PR is merged)

**Comment by holgerd77** [comment]
This would need a cherry-picked re-submission against the latest rebased `develop` from #1873.

**Comment by ryanio** [comment]
phew, I have rebased this on latest develop :) ready for final review and merging, let's do it!

File: packages/block/src/block.ts
Comment (acolytec3) [inline packages/block/src/block.ts:326]: This is a total nit but should we replace 'uh' with `uncleHeader` just to make it clear.  Sometimes these short acronyms aren't obvious.  My first glance at this had me wondering "uh...what is 'uh'!?'"

File: packages/block/test/block.spec.ts
Comment (ryanio) [reply packages/block/test/block.spec.ts:723]: have updated, thx

File: packages/block/src/block.ts
Comment (acolytec3) [reply packages/block/src/block.ts:326]: Works for me.

File: packages/blockchain/src/db/helpers.ts
Comment (ryanio) [reply packages/blockchain/src/db/helpers.ts:47]: It's something we've talked about at length. I think it's the direction we want to go, but all our APIs support Buffers so it would take a lot of work to update, probably a similar effort level of the BN.js->bigint change. Buffers use Uint8Arrays  as their data type so I don't believe it's a big deal to make a Buffer out of a Uint8Array. We could do some benchmarking to confirm that it is relatively quick. In general we can move away from Buffers because they provide very little added benefit as opposed to just working with Uint8Arrays directly, as we've done with `rlp` and as `js-ethereum-cryptography` has done switching primarily from Buffers to Uint8Arrays. Also gives us broader ecosystem support since you have to polyfill Buffer in browser and other contexts.

**Comment by acolytec3** [comment]
Looks good overall.  One question more generally related to `rlp`, what would you think about moving the two helpers `buffArrToArr` and `arrToBuffArr` into the rlp module?  Feels like they are generally specifically used for RLP deserialization so could be useful to package them with RLP rather than `util` since that means projects that are just using `rlp` also need to bundle `ethereumjs-util` with their code to make full use of RLP.

**Review by acolytec3 — APPROVED** [review]
LGTM!  Nice work and a long slog I'm sure.

File: packages/devp2p/src/rlpx/ecies.ts
Comment (ryanio) [reply packages/devp2p/src/rlpx/ecies.ts:335]: This is unrelated to this PR, but I took a look, these are the `header-data` fields from [rlpx.md#framing](https://github.com/ethereum/devp2p/blob/master/rlpx.md#framing): `header-data = [capability-id, context-id]`

> The frame encoding provides `capability-id` and `context-id` fields for multiplexing purposes, but these fields are unused.`

So it's not a TODO, I'll improve in a commit.

**Review by acolytec3 — APPROVED** [review]
LGTM, pending tests passing!

**Review by holgerd77 — APPROVED** [review]
Nice, yes, LGTM (only superficial second look), and it' really nice to see the `externals` module finally completely removed in `Utils` as well! 💯 😀 🙏

Will merge this in.

**Pull request merged**
